emitters/yaml_base_emitter.cpp

```diff
@@ -442,8 +675,13 @@ bool yaml_base_emitter::check_object_array(const std::string& filepath,
     };
 
     if (!expected_node.count(key)) {
```

Comment:
This check looks like it could be generalized into a routine and called from the multiple `check_*` APIs.